internal/dispatch/graph/graph.go

```diff
@@ -154,28 +154,12 @@ func (ld *localDispatcher) lookupRelation(ctx context.Context, ns *core.Namespac
 	return relation, nil
 }
 
-type stringableOnr struct {
-	*core.ObjectAndRelation
-}
-
-func (onr stringableOnr) String() string {
-	return tuple.StringONR(onr.ObjectAndRelation)
-}
-
-type stringableRelRef struct {
-	*core.RelationReference
-}
-
-func (rr stringableRelRef) String() string {
-	return fmt.Sprintf("%s::%s", rr.Namespace, rr.Relation)
-}
-
 // DispatchCheck implements dispatch.Check interface
 func (ld *localDispatcher) DispatchCheck(ctx context.Context, req *v1.DispatchCheckRequest) (*v1.DispatchCheckResponse, error) {
 	ctx, span := tracer.Start(ctx, "DispatchCheck", trace.WithAttributes(
-		attribute.Stringer("resource-type", stringableRelRef{req.ResourceRelation}),
+		attribute.String("resource-type", tuple.StringRR(req.ResourceRelation)),
```

Comment:
Downside of this change is that we'll always be creating the string form of the resource relation

Reply:
True, what if I made the Stringer struct generic, so we could at least reuse it other places?

```go
type CustomStringer[T any] struct {
  fn func(T) string
  T
}

func (cs *CustomStringer[T]) String() string { return cs.fn(cs.T) }
```

Reply:
Sounds good to me

Reply:
On second thought, that'd put `fn` on the heap. I think since we always ultimately do create the string, doing what's in the PR is fine.